CHANGES.md

```diff
@@ -13,14 +13,35 @@ and this project adheres to [Semantic Versioning](https://semver.org/).
 
 - A small tutorial in tests/OK/tuto.lp.
 
+#### Type checker with refinement
+
+Implemented a refiner, a type checker which is able to edit the type checked
+term. The algorithm is taken from <https://arxiv.org/abs/1202.4905>.
+
+- New type checker (with refinement)
+- Added placeholders. Placeholders are the holes in the concrete syntax, they
+  are transformed into metavariables by the type checker. It simplifies some
+  code related to metavariables. *Placeholders cannot appear nonlinearly in a
+  term*.
+- No more unused variable warning (such variables may appear in the explicit
+  substitution of a metavariable).
+ 
```

Comment:
There is still some code generating warnings:
```
16:24 ~/src/lambdapi (refiner-merge-steps) gl 'could be replaced'
./src/handle/command.ml:402:                    wrn pos "Variable [%a] could be replaced by [_]."
./src/parsing/scope.ml:307:            wrn pos "Variable %s could be replaced by '_'." id;
./src/parsing/scope.ml:381:              "Pattern [%a] could be replaced by [_]." Pretty.term t;
```

Reply:
We keep warning for unused patterns